### PR TITLE
Port transaction pool benchmark

### PR DESCRIPTION
**Pull request by skunert**
This PR provides a benchmark similar to the one in [substrate](https://github.com/paritytech/substrate/blob/master/bin/node/cli/benches/transaction_pool.rs).

The benchmark 
- starts two relay chain nodes and two parachain nodes
- registers the parachain
- initializes `account_num` accounts on the parachain
- performs `extrinsics_per_account` transfers per account

Current results on MBP M1:
```
time:   [17.170 s 20.712 s 24.183 s]
thrpt:  [8.2704  elem/s 9.6561  elem/s 11.648  elem/s]
```

### CODE REVIEW

File: test/service/benches/transaction_pool.rs [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+	// Run dave as parachain collator
+	let dave = runtime.block_on(
+		cumulus_test_service::TestNodeBuilder::new(para_id, tokio_handle.clone(), Dave)
+			.enable_collator()
+			.connect_to_parachain_node(&charlie)
+			.connect_to_relay_chain_nodes(vec![&alice, &bob])
+			.build(),
+	);
+
+	runtime.block_on(dave.wait_for_blocks(2));
```
Comment (bkchr) [inline test/service/benches/transaction_pool.rs:197]:
```suggestion
	runtime.block_on(dave.wait_for_blocks(1));
```

File: test/service/src/lib.rs [hunk 2]
```diff
@@ -669,17 +677,27 @@ impl TestNode {
 	}
 }
 
+/// Fetch account nonce for key pair
+pub fn fetch_nonce(client: &Client, account: sp_core::sr25519::Pair) -> u32 {
```
Comment (bkchr) [inline test/service/src/lib.rs:681]:
```suggestion
pub fn fetch_nonce(client: &Client, account: sp_core::sr25519::Public) -> u32 {
```

File: test/service/src/lib.rs [hunk 3]
```diff
@@ -669,17 +677,27 @@ impl TestNode {
...
+	let best_hash = client.chain_info().best_hash;
+	client
+		.runtime_api()
+		.account_nonce(&generic::BlockId::Hash(best_hash), account.public().into())
```
Comment (bkchr) [inline test/service/src/lib.rs:685]:
```suggestion
		.account_nonce(&generic::BlockId::Hash(best_hash), account.into())
```

File: test/service/src/lib.rs [hunk 4]
```diff
@@ -669,17 +677,27 @@ impl TestNode {
...
-	caller: Sr25519Keyring,
+	caller: sp_core::sr25519::Pair,
+	nonce: Option<u32>,
 ) -> runtime::UncheckedExtrinsic {
 	let function = function.into();
 	let current_block_hash = client.info().best_hash;
 	let current_block = client.info().best_number.saturated_into();
 	let genesis_block = client.hash(0).unwrap().unwrap();
-	let nonce = 0;
+	let nonce = nonce.unwrap_or_else(|| fetch_nonce(client, caller.clone()));
```
Comment (bkchr) [inline test/service/src/lib.rs:700]:
```suggestion
	let nonce = nonce.unwrap_or_else(|| fetch_nonce(client, caller.public()));
```

File: test/runtime-upgrade/Cargo.toml [hunk 5]
```diff
@@ -13,6 +13,7 @@ serde = { version = "1.0.101", optional = true, features = ["derive"] }
 frame-executive = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "master" }
 frame-support = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "master" }
 frame-system = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "master" }
+frame-system-rpc-runtime-api = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "master" }
```
[the remark this answers is not in the document]
Comment (skunert) [reply test/runtime-upgrade/Cargo.toml:16]: Did you intend to and it slipped through? Looks like the file is still there: https://github.com/paritytech/cumulus/blob/master/test/runtime-upgrade/Cargo.toml

[the remark this answers is not in the document]
Comment (bkchr) [reply test/runtime-upgrade/Cargo.toml:16]: Hmm yes, pelase remove it.

**Review by bkchr — APPROVED** [review]

**Pull request merged**